### DISCUSSION TITLE
Is there a way to use MAUI Handler without registering with AddHandler?

### DISCUSSION THREAD

**Discussion by qq570850096** [dotnet/maui#14739] [Q&A]
When using MAUI, if MAUI's own controls, such as Entry, do not need to be registered, then in which part of the source code are they registered? Is there a way to use this approach in your own custom controls to avoid registration with AddHandler?

**Comment by FM1973** [comment]
I think this is possible.
As far as I know, you can access the handler of a control at any time.
I.e. you can access the handler of a picker control in codebehind like this:

`if (picker.Handler is IDatePickerHandler handler) handler.xxxx`

Hope this helps.

**Comment by xtuzy** [comment]
Maui's control all use Handler, see 

https://github.com/dotnet/maui/blob/3518fc085432a41476018d67cb9d024e70a1fadd/src/Controls/src/Xaml/Hosting/AppHostBuilderExtensions.cs#L61

if you have a custom android AppCompatEditText control, you can inherit EntryHandler
https://github.com/dotnet/maui/blob/3518fc085432a41476018d67cb9d024e70a1fadd/src/Core/src/Handlers/Entry/EntryHandler.Android.cs#L14
override CreatePlatformView() and use your custom android AppCompatEditText to create it

finally, use `.AddHandler<Entry, YourEntryHandler>();`
